base/event.jl

```diff
@@ -150,19 +150,64 @@ tasks.
 yield() = (enq_work(current_task()); wait())
 
 """
+    yield(t::Task, arg = nothing)
+
+A fast, unfair-scheduling version of `schedule(t, arg); yield()` which
+immediately yields to `t` before calling the scheduler.
+"""
+function yield(t::Task, x::ANY = nothing)
+    t.state == :runnable || error("schedule: Task not runnable")
+    t.result = x
+    enq_work(current_task())
+    return try_yieldto(ensure_self_descheduled, t)
+end
+
+"""
     yieldto(t::Task, arg = nothing)
 
 Switch to the given task. The first time a task is switched to, the task's function is
 called with no arguments. On subsequent switches, `arg` is returned from the task's last
 call to `yieldto`. This is a low-level call that only switches tasks, not considering states
 or scheduling in any way. Its use is discouraged.
 """
-yieldto(t::Task, x::ANY = nothing) = ccall(:jl_switchto, Any, (Any, Any), t, x)
+function yieldto(t::Task, x::ANY = nothing)
+    t.result = x
+    return try_yieldto(Void, t)
+end
+
+function try_yieldto(undo::F, t::Task) where F
```

Comment:
I think @tkelman is right here. There are a lot of type signatures that end up having unnecessary type parameters for various reasons, which we tend to reflexively remove. ANY only gets used for a reason, which we don't reflexively use. So unless you comment this or test it somehow, you should expect that this will get deleted at some point.